docs/source-pytorch/cli/lightning_cli_advanced_3.rst

```diff
@@ -354,3 +354,33 @@ You can also pass the class path directly, for example, if the optimizer hasn't
         --optimizer1.lr=0.01 \
         --optimizer2=torch.optim.AdamW \
         --optimizer2.lr=0.0001
+
+
+Run from Python
+^^^^^^^^^^^^^^^
+
+Even though the :class:`~pytorch_lightning.cli.LightningCLI` class is designed to help in the implementation of command
+line tools, for some use cases it is desired to run directly from Python. To support these use cases, the ``args``
+parameter can be used, for example:
+
+.. testcode::
+
+    cli = LightningCLI(MyModel, args=["--trainer.max_epochs=100", "--model.encoder_layers=24"])
```

Comment:
To clarify, are these merged with the normal CLI args? And `--config=...`? If so, which takes precedence?

Reply:
They are not merged. If `args != None` and this is used from the command line, all the command line arguments given would be ignored. `args` is only to be able to run things from Python like if it were from the command line.

Reply:
In that case, can we check if command line args were also passed and then raise an error?

I imagine people will expect it to work and would be confused if they are ignored.

Reply:
Good idea!